### PR TITLE
LibWeb: Add workaround to restore cross-origin stylesheet functionality

### PR DESCRIPTION
**Pull request by Lubrsi**
CORS cross-origin responses in the No CORS request mode provide an opaque filtered response, which is the original response with certain attributes removed/changed.

The relevant effect it has is setting the body to `null`, which means `body_bytes` has `Empty` in the process_response_consume_body callback. This effectively disables cross-origin linked resources (e.g. stylesheets).

However, the web actually depends on this, especially for stylesheets retrieved from a cross-origin CDN. For example, Shopify websites request stylesheets from `cdn.shopify.com` and Substack websites request stylesheets from `substackcdn.com`.

This makes this a specification bug, as this code was written from it.

The workaround is to read the actual body from the unfiltered response and then call `process_linked_resource` from there.

This _should_ be safe to do, as linked resource fetches do not include credentials (i.e. cookies and the Authorization header), so it cannot provide personalized responses.

### CODE REVIEW

File: Userland/Libraries/LibWeb/HTML/HTMLLinkElement.cpp [hunk 1]
```diff
@@ -294,8 +294,48 @@ void HTMLLinkElement::default_fetch_and_process_linked_resource()
...
+                        // CORS cross-origin responses in the No CORS request mode provide an opaque filtered response, which is the original response
+                        // with certain attributes removed/changed.
+
+                        // The relevant effect it has is setting the body to `null`, which means `body_bytes` has `Empty` here. This effectively
+                        // disables cross-origin linked resources (e.g. stylesheets).
+
+                        // However, the web actually depends on this, especially for stylesheets retrieved from a cross-origin CDN. For example,
+                        // Shopify websites request stylesheets from `cdn.shopify.com` and Substack websites request stylesheets from `substackcdn.com`.
+
+                        // This makes this a specification bug, as this code was written from it.
```
Comment (ADKaster) [inline Userland/Libraries/LibWeb/HTML/HTMLLinkElement.cpp:307]: Feels like we should be good bois and report the spec bug to WHATWG, or link one if there's already an existing issue.

Comment (Lubrsi) [reply Userland/Libraries/LibWeb/HTML/HTMLLinkElement.cpp:307]: Done: https://github.com/whatwg/html/issues/9066

Comment (Lubrsi) [reply Userland/Libraries/LibWeb/HTML/HTMLLinkElement.cpp:307]: Also added to the comment

**Review by linusg — APPROVED** [review]
:100:, thanks for filing a spec issue as well!

**Pull request merged**